pkg/controllers/networkpolicy/controller_test.go

```diff
@@ -433,36 +432,60 @@ var _ = Describe("NetworkPolicy", func() {
 		}
 	}
 
-	// The sequence of IPs in IngressRule/EgressRule might change during rule deduplication.
-	// Sort it for equality check.
-	sortRuleIPs := func(inputIP []*net.IPNet) []*net.IPNet {
-		if inputIP == nil {
-			return nil
+	checkRules := func(grpID *securitygroup.CloudResource, add, rm []*securitygroup.CloudRule) {
+		addIngress := make([]securitygroup.IngressRule, 0)
+		addEgress := make([]securitygroup.EgressRule, 0)
+		for _, rule := range add {
+			switch rule.Rule.(type) {
+			case *securitygroup.IngressRule:
+				addIngress = append(addIngress, *rule.Rule.(*securitygroup.IngressRule))
+			case *securitygroup.EgressRule:
+				addEgress = append(addEgress, *rule.Rule.(*securitygroup.EgressRule))
+			}
 		}
-		ipMap := make(map[string]*net.IPNet)
-		ipStrList := make([]string, 0)
-		for _, ip := range inputIP {
-			ipStrList = append(ipStrList, ip.String())
-			ipMap[ip.String()] = ip
+		rmIngress := make([]securitygroup.IngressRule, 0)
+		rmEgress := make([]securitygroup.EgressRule, 0)
+		for _, rule := range rm {
```

Comment:
we have a new function for this right? splitCloudRulesByDirection

Reply:
Done